### PR TITLE
Amend 2nd Firefox launch via xterm to avoid password input lose

### PR DESCRIPTION
**Pull request by jouyingbin**
**Description**:

The problem is randomly reproduced in aarch64 and few times in x86_64/s390x, I suspect the error is also a race condition about poor worker performance in aarch64. Plan to extend a bit the waiting time and also adjust the 2nd Firefox launch approach. It's probably also some trouble for **x11_start_program**() if inputting 3 parameters under the bad internet connection, thus the input password pop-up window is not reacted before **type_string $fips_password** and cause the password input lose.

1. Replace x11_start_program 2nd Firefox launch to xterm
2. Add 2 needles match to avoid racing condition

- Related ticket: https://progress.opensuse.org/issues/75283
- Needles: 2 Needles are merged in O.S.D.
- Verification run: 
https://openqa.suse.de/t4899112 (x86_64)
https://openqa.suse.de/t4899161 (s390x)
https://openqa.suse.de/tests/4913843 (aarch64)

### CODE REVIEW

**Review by rfan1 — COMMENTED** [review]
# Close Firefox
    quit_firefox;
    assert_screen "generic-desktop";

}
Ben, why no workaround here?

**Comment by jouyingbin** [comment]
> ```
> # Close Firefox
> quit_firefox;
> assert_screen "generic-desktop";
> ```
> 
> }
> Ben, why no workaround here?

I think there is no need to add more wait time here from my testing. The real problem is before the 2nd Firefox launch from x11_start_problem(), so I add some wait time before it.

**Comment by rfan1** [comment]
> > ```
> > # Close Firefox
> > quit_firefox;
> > assert_screen "generic-desktop";
> > ```
> > 
> > 
> > }
> > Ben, why no workaround here?
> 
> I think there is no need to add more wait time here from my testing. The real problem is before the 2nd Firefox launch from x11_start_problem(), so I add some wait time before it.

    # Close Firefox
    quit_firefox;
    wait_still_screen 10;
    assert_screen "generic-desktop";
    wait_still_screen 10;

I am interested in this, you have added some delay here, if no issue seen later, would be no issue seen here, at least the 1st "    wait_still_screen 10;" is not a must one

**Review by rfan1 — APPROVED** [review]
LGTM

**Review by lilyeyes — APPROVED** [review]
LGTM

File: tests/fips/mozilla_nss/firefox_nss.pm [hunk 1]
```diff
@@ -80,36 +79,52 @@ sub run {
...
     # "start_firefox" will be not used, since the master password is
     # required when firefox launching in FIPS mode
-    x11_start_program('firefox --setDefaultBrowser https://html5test.opensuse.org', target_match => 'firefox-fips-password-inputfiled', match_timeout => 360);
-
+    # x11_start_program('firefox --setDefaultBrowser https://html5test.opensuse.org', target_match => 'firefox-fips-password-inputfiled', match_timeout => 180);
+    # Adjust the 2nd Firefox launch via xterm as workaround to avoid the race condition during password interaction while internet connection busy
+    x11_start_program('xterm');
+    mouse_hide(1);
+    type_string("firefox --setDefaultBrowser https://html5test.opensuse.org\n");
+    wait_still_screen 120;
```
Comment (lemon-suse) [inline tests/fips/mozilla_nss/firefox_nss.pm:96]: Maybe you can use assert_screen $timeout instead the wait_still_screen, I don't think it is suitable here.

Comment (jouyingbin) [reply tests/fips/mozilla_nss/firefox_nss.pm:96]: thanks for the comment, I will change it.

**Comment by lemon-suse** [comment]
LGTM.

**Pull request merged**